packages/transformers/svelte/src/SvelteTransformer.js

```diff
@@ -0,0 +1,147 @@
+import {Transformer} from '@parcel/plugin';
+import SourceMap from '@parcel/source-map';
+import {relativeUrl} from '@parcel/utils';
+import {compile, preprocess} from 'svelte/compiler';
+import ThrowableDiagnostic from '@parcel/diagnostic';
+
+export default new Transformer({
+  async loadConfig({config, options, logger}) {
+    const svelteConfig = await config.getConfig([
+      '.svelterc',
+      'svelte.config.js',
+    ]);
+    if (!svelteConfig) return {};
+    if (svelteConfig.filePath.endsWith('.js')) {
+      // TODO: Is there a better way of handling this warning? Probably just
+      // mention it in the documentation and silently invalidate.
+      logger.warn({
+        message:
+          'WARNING: Using a JavaScript Svelte config file means losing ' +
+          'out on caching features of Parcel. Use a .svelterc(.json) ' +
+          'file whenever possible.',
+      });
+      config.invalidateOnStartup();
+    }
+    return {
+      ...svelteConfig.contents,
+      compilerOptions: {
+        // TODO: Call this out in the documentation.
+        css: false,
+        ...svelteConfig.contents.compilerOptions,
+        dev: options.mode !== 'production',
+      },
+    };
+  },
+
+  async transform({asset, config, options, logger}) {
+    let source = await asset.getCode();
+    const filename = relativeUrl(options.projectRoot, asset.filePath);
+
+    if (!config.preprocess) {
+      let preprocessor = null;
+      try {
+        // This seems sufficiently batteries included. Only apply
+        // svelte-preprocess if it is importable. Otherwise, assume the lack of
+        // a preprocess entry is intentional.
+        // TODO: Is it possible to utilize parcel's pipelines in lieu of
+        // svelte-preprocess?
+        // TODO: Call this out in the documenation.
+        logger.verbose({message: 'Attempting to use svelte-preprocess.'});
+        preprocessor = await import('svelte-preprocess');
```

Comment:
Note: using ESM currently means Parcel cannot invalidate the cache properly when this dependency changes.

Reply:
Can I just `require('svelte-preprocess')` instead?

Reply:
That was probably an unnecessary question... I fixed it in the gcoakes/parcel@d896bfd.

Reply:
I'm still uncertain how to write an integration test for the negative case of this import. It seems like the test harness isn't designed for optional dependencies. I think it's fairly safe to leave untested though as long as the catch block remains minimal.